webform_confirm_email.module

```diff
@@ -376,7 +376,7 @@ function webform_confirm_email_tokens($type, $tokens, array $data = array(), arr
   }
   $node = isset($data['node']) ? $data['node'] : node_load($submission->nid);
   if (   $type === 'submission'
-      && isset($tokens['confirm_url'])
+      && (isset($tokens['confirm_url']) || isset($tokens['confirm_url:raw']))
```

Comment:
Ah, nice that you found the cause of the bug.

I would prefer removing the check for `$tokens['confirm_url']` instead of extending it to another name. The code here is already far too complex for what it does.

Reply:
Ok, changed it to only check for tokens in general.

Reply:
I would even remove that. If `$token` was empty then this function would never be called with `$type = 'submission';`.

Reply:
If you say so → have removed it.